docs/source/x_ext.rst

```diff
@@ -256,6 +256,24 @@ The ``id`` values for in-flight offloaded instructions are required to be unique
 The ``id`` values are required to be incremental from one issue transaction to the next.
 The increment may be greater than one.
 If the next ``id`` would be greater than the maximum value (``2**X_ID_WIDTH - 1``), the value of ``id`` wraps.
+A new ``id`` value is not allowed to be greater than the oldest in-flight instruction.
+If the oldest in-flight instruction is :math:`id_o`, and the newest is ``id_n``, then the next instruction with ``id_n+1`` must satisfy the following conditions:
+
+.. math::
+  \begin{gather}
+    id_{n+1} > id_{n} \text{ or } id_{n+1} < id_{o}, \text{ if } id_{n} > id_{o}\\
+    id_{n+1} > id_{n} \text{ and } id_{n+1} < id_{o}, \text{ if } id_{n} < id_{o}
+  \end{gather}
+
+The first condition applying to cases where the ``id`` has wrapped since the oldest in-flight instruction was issued, and the second where no wrap occurred.
+
+
+.. note::
+  IDs are not required to be incremental to support scenarios, in which a coprocessor does not see the entire instruction stream.
+  This can be e.g. because offloaded instructions are routed towards different |coprocessors|.
```

Comment:
A coprocessor will only see well ordered instructions. Gaps may be there.

Reply:
_I can be wrong. But as far as I understand the interface protocol:_

Because of this exception, the coprocessor cannot make any assumption on the incremental property of the IDs it receives. So it can only rely on the order of issued instructions to correctly kill in-flight instruction when it receives a commit_kill.
Then there is no reason to put any constraint on IDs in this _interface specification_ other than the fact that all in-flight IDs need to be different, so that the CV-X-IF handshaking is correct.
The generation of IDs is then up to the processor designer; good practices could be recommended in the Appendix, but do not look necessary in the normative section.

Reply:
I was under the same impression until yesterday, where I understood it like this:

With the new addition of ID_WIDTH being _large enough_, from coprocessor side we at least know that the incoming IDs are not "jumping" in the sense of 0, 4, 2, 5, 3, ... (with the exception of wrap-around when maxing out the ID count).
This may help in the coprocessor design when killing or committing instructions in the middle of the current in-flight instruction chain?

I could be wrong too though, it is more of a feeling than thought through.

Reply:
> _I can be wrong. But as far as I understand the interface protocol:_
> 
> Because of this exception, the coprocessor cannot make any assumption on the incremental property of the IDs it receives. So it can only rely on the order of issued instructions to correctly kill in-flight instructions when it receives a commit_kill. Then there is no reason to put any constraint on IDs in this _interface specification_ other than the fact that all in-flight IDs need to be different, so that the CV-X-IF handshaking is correct. The generation of IDs is then up to the processor designer; good practices could be recommended in the Appendix, but do not look necessary in the normative section.

It makes a big difference. If they are random, you need to build a lookup. If they are ordered, you can compare. 
For a piece of hardware, this whole thing is actually easier than for a human brain, because any math it would be doing would also wrap with the boundaries of ID_WIDTH.